src/java/org/apache/cassandra/db/compression/CompressionDictionaryManager.java

```diff
@@ -0,0 +1,346 @@
+
+                trainer = ICompressionDictionaryTrainer.create(keyspaceName, tableName, newParams, createTrainingConfig());
+                trainer.setDictionaryTrainedListener(this::handleNewDictionary);
+            }
+
+            scheduler.scheduleRefreshTask();
+
+            // Start trainer if it exists
+            if (trainer != null)
+            {
+                trainer.start(false);
+            }
+            return;
+        }
+
+        // Clean up when dictionary compression is disabled
+        try
+        {
+            close();
+        }
+        catch (Exception e)
+        {
+            logger.warn("Failed to close CompressionDictionaryManager on disabling " +
+                        "dictionary-based compression for table {}.{}", keyspaceName, tableName);
+        }
+    }
+
+    /**
+     * Adds a sample to the dictionary trainer for learning compression patterns.
+     * Samples are randomly selected to avoid bias and improve dictionary quality.
+     *
+     * @param sample the sample data to potentially add for training
+     */
+    public void addSample(ByteBuffer sample)
+    {
+        ICompressionDictionaryTrainer dictionaryTrainer = trainer;
+        if (dictionaryTrainer != null && dictionaryTrainer.shouldSample())
+        {
+            dictionaryTrainer.addSample(sample);
+        }
+    }
+    
+    @Nullable
+    @Override
+    public CompressionDictionary getCurrent()
+    {
+        return cache.getCurrent();
+    }
+
+    @Override
+    public CompressionDictionary get(CompressionDictionary.DictId dictId)
+    {
+        return cache.get(dictId);
+    }
+
+    @Override
+    public void add(CompressionDictionary compressionDictionary)
+    {
+        cache.add(compressionDictionary);
+    }
+
+    @Override
+    public void setCurrentIfNewer(@Nullable CompressionDictionary dictionary)
+    {
+        cache.setCurrentIfNewer(dictionary);
+    }
+
+    @Override
+    public void onNewDictionaryTrained(long dictionaryId)
+    {
+        eventHandler.onNewDictionaryTrained(dictionaryId);
+    }
+
+    @Override
+    public void onNewDictionaryAvailable(long dictionaryId)
+    {
+        eventHandler.onNewDictionaryAvailable(dictionaryId);
+    }
+
+    @Override
+    public synchronized void train(Map<String, String> options)
+    {
+        // Validate table supports dictionary compression
+        if (!isEnabled)
+        {
+            throw new IllegalArgumentException("Table " + keyspaceName + '.' + tableName + " does not support dictionary compression");
+        }
+
+        if (trainer == null)
+        {
+            throw new IllegalStateException("Dictionary trainer is not available for table " + keyspaceName + '.' + tableName);
+        }
+
+        // Parse and validate training options
+        ManualTrainingOptions trainingOptions = ManualTrainingOptions.fromStringMap(options);
+
+        trainer.start(true);
+        scheduler.scheduleManualTraining(trainingOptions, trainer);
```

Comment:
we would not actually _scheduled_ anything in that method if trainer is no-op instance.